helix-term/src/application.rs

```diff
@@ -98,30 +101,34 @@ impl Application {
         });
         let syn_loader = std::sync::Arc::new(syntax::Loader::new(syn_loader_conf));
 
+        let config = Arc::new(ArcSwap::from_pointee(config));
         let mut editor = Editor::new(
             size,
             theme_loader.clone(),
             syn_loader.clone(),
-            config.editor.clone(),
+            Box::new(Map::new(Arc::clone(&config), |config: &Config| {
+                &config.editor
+            })),
         );
 
-        let editor_view = Box::new(ui::EditorView::new(std::mem::take(&mut config.keys)));
+        let keymaps = Box::new(Map::new(Arc::clone(&config), |config: &Config| {
+            &config.keys
+        }));
```

Comment:
Can we avoid boxing by using `Map<>` directly? For that to work you will probably need to use a `fn` as the last parameter rather than a closure, so that you can name it on the type